source/simulator/parameters.cc

```diff
@@ -937,6 +972,26 @@ namespace aspect
         stabilization_c_R                   = prm.get_double ("cR");
         stabilization_beta                  = prm.get_double ("beta");
         discontinuous_penalty               = prm.get_double ("Discontinuous penalty");
+        use_limiter_for_discontinuous_temperature_solution
+          = prm.get_bool("Use limiter for discontinuous temperature solution");
+        use_limiter_for_discontinuous_composition_solution
+          = prm.get_bool("Use limiter for discontinuous composition solution");
+        if (use_limiter_for_discontinuous_temperature_solution
+            || use_limiter_for_discontinuous_composition_solution)
+          AssertThrow (nonlinear_solver == NonlinearSolver::IMPES,
+                       ExcMessage ("The bound preserving limiter currently is "
+                                   "only implemented for the scheme using IMPES nonlinear solver. "
+                                   "Please deactivate the limiter or change the solver scheme."));
+        global_temperature_max_preset       = prm.get_double ("Global temperature max");
+        global_temperature_min_preset       = prm.get_double ("Global temperature min");
+        const std::vector<double> composition_max_fields = Utilities::string_to_double
+                                                           (Utilities::split_string_list(prm.get ("Global composition max")));
+        global_composition_max_preset       = std::vector<double> (composition_max_fields.begin(),
```

Comment:
It's not quite clear to me why you need the `composition_max_fields` variable here. Why not just say
```
  global_composition_max_preset = Utilities::string_to_double (...);
```
?

Also, can you add an assertion that the size of this array is really equal to the number of compositional fields?